File: prototype/rails/src/test/domain/cart/cart_line_test.rb
require "test_helper"

module Domain
  module Cart
    class CartLineTest < ActiveSupport::TestCase
      test "a line totals its unit price" do
        line = CartLine.new(seller_id: 1, unit_price: Money.from_cents(4500), quantity: 3)

        assert_equal 13_500, line.total.cents
      end

      test "a line covers at least one item" do
        assert_raises(ArgumentError) { CartLine.new(seller_id: 1, unit_price: Money.from_cents(4500), quantity: 0) }
      end
    end
  end
end

File: prototype/php/src/app/Domain/Cart/CartLineTest.php
<?php

declare(strict_types=1);

namespace App\Domain\Cart;

use App\Domain\Money\Money;
use InvalidArgumentException;

it('totals its unit price across the quantity', function (): void {
    $line = CartLine::of(1, Money::fromCents(4500), 3);

    expect($line->total()->cents)->toBe(13500);
});

it('needs at least one item', function (): void {
    expect(fn () => CartLine::of(1, Money::fromCents(4500), 0))
        ->toThrow(InvalidArgumentException::class);
});
